### PR TITLE
feat: add references and utils for delete storage key

### PR DESCRIPTION
**Pull request by Jonathansoufer**
## **Description**

This PR implements a feature for our users get access to delete their Notification's storage keys if need.

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to Settings
2. Choose Notifications
3. Make sure its enabled
4. Click on Reset Notifications
5. Proceed 
6. Check if Notifications remain enabled and a toast is shown.

### **After**

https://github.com/user-attachments/assets/07ed0870-a6f2-442c-a692-880037f03f8d

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: locales/languages/en.json [hunk 1]
```diff
@@ -770,8 +770,9 @@
     "enabling_notifications": "Enabling notifications...",
     "disabling_notifications": "Disabling notifications...",
     "reset_notifications_title": "Reset notifications",
-    "reset_notifications_description": "Resetting notifications, means you're overwritting your notifications storage keys and resetting all your notification history. Are you sure you want to do this?",
+    "reset_notifications_description": "Resetting notifications, means you're deleting your notifications storage keys and resetting all your notification history. Are you sure you want to do this?",
```
Comment (tommasini) [inline locales/languages/en.json:773]: Hmm, since this is not creating a new key, I would keep an eye to see if this is translated, I'm not sure

**Review by tommasini — APPROVED** [review]

**Comment by tommasini** [comment]
Can you add manual steps to the PR description please? It will be easier to test on release that way!

File: locales/languages/en.json (hunk 1 above)
Comment (Jonathansoufer) [reply locales/languages/en.json:773]: Good one!

File: app/actions/notification/helpers/index.ts [hunk 2]
```diff
@@ -171,3 +171,20 @@ export const markMetamaskNotificationsAsRead = async (
...
   }
 };
+/**
+ * Perform the deletion of the notifications storage key and the creation of on chain triggers to reset the notifications.
+ *
+ * @returns {Promise<string | undefined>} A promise that resolves to a string error message or undefined if successful.
+ */
+export const performDeleteStorage = async (): Promise<string | undefined> => {
+  try {
+   await Engine.context.UserStorageController.performDeleteStorage('notifications.notification_settings');
```
Comment (Prithpal-Sooriya) [inline app/actions/notification/helpers/index.ts:181]: Wow this is cool!

File: app/components/UI/Notification/ResetNotificationsModal/__snapshots__/ResetNotificationsModal.test.tsx.snap [hunk 3]
```diff
@@ -170,7 +170,7 @@ exports[`ProfileSyncingModal should render correctly 1`] = `
             }
           }
         >
-          Resetting notifications, means you're overwritting your notifications storage keys and resetting all your notification history. Are you sure you want to do this?
+          Resetting notifications, means you're deleting your notifications storage keys and resetting all your notification history. Are you sure you want to do this?
```
Comment (Prithpal-Sooriya) [inline app/components/UI/Notification/ResetNotificationsModal/__snapshots__/ResetNotificationsModal.test.tsx.snap:173]: Yep nice!

**Review by Prithpal-Sooriya — APPROVED** [review]

**Review by EtherWizard33 — APPROVED** [review]

**Pull request merged**